mlir/include/mlir/Dialect/WebAssembly/IR/WebAssemblyOps.td

```diff
@@ -565,6 +565,38 @@ def Wasm_ConvertSOp : Wasm_ConversionOp<"convert_s",
     [Wasm_IntegerType],
     [Wasm_FPType]>{}
 
+def Wasm_ExtendSI32Op : Wasm_Op<"extend_i32_s">{
+  let summary = [{"Extend i32 to i64 performing sign extension."}];
+  let arguments = (ins I32:$input);
+  let results = (outs I64:$result);
+  let assemblyFormat = "$input `to` type($result)  attr-dict";
+}
+
+def Wasm_ExtendUI32Op : Wasm_Op<"extend_i32_u">{
```

Comment:
Why not reuse the ConversionOp class for those two?

Reply:
Because we already have the type of input in the op mnemonic, so I basically wanted to avoid having it in the assembly format too. 

Maybe I can make this an option of the conversionOp class, but it's not clear to me how to do it at the moment.

Reply:
Ok I understand no problem